service/ui.go

```diff
@@ -446,5 +444,15 @@ func (uis *UIServer) GetServiceApp() *gimlet.APIApp {
 	app.PrefixRoute("/plugin").Route("/buildbaron/note/{task_id}").Wrap(needsLogin, needsContext, viewTasks).Handler(bbGetNote).Get()
 	app.PrefixRoute("/plugin").Route("/buildbaron/note/{task_id}").Wrap(needsLogin, needsContext, editTasks).Handler(bbSaveNote).Put()
 	app.PrefixRoute("/plugin").Route("/buildbaron/file_ticket").Wrap(needsLogin, needsContext).Handler(uis.bbFileTicket).Post()
+
+	// Add an OPTIONS method to every POST request to handle preflight OPTIONS requests.
+	// These requests must not check for credentials. They exist to validate whether a route exists, and to
+	// allow requests from an origin.
+	for _, r := range app.Routes() {
+		if r.HasMethod(http.MethodPost) {
+			app.AddRoute(r.GetRoute()).Wrap(allowsCORS).Handler(func(_ http.ResponseWriter, _ *http.Request) {}).Options()
```

Comment:
My question about the 200 response was because this seems to return no response to the request. Is this handler supposed to `gimlet.WriteJSON(w, "")`?

Reply:
Oh, it seemed to already return 200 when I checked in the network tab 🤔

<img width="476" alt="Screenshot 2023-04-25 at 11 11 50 AM" src="https://user-images.githubusercontent.com/47064971/234323661-62708bc1-0940-42df-a508-a4374d334928.png">

I updated it to use `gimlet.WriteJSON(w, "")` and it still works so that sounds good to me!